src/solve/test.rs

```diff
@@ -175,7 +175,7 @@ fn prove_forall() {
         // Here, we do know that `T: Clone`, so we can.
         goal {
             forall<T> {
-                if (T: Clone) {
+                if (WellFormed(T: Clone), T: Clone) {
```

Comment:
Yes. So I wonder if it's worth having two forms -- e.g., `if` and `if-raw` or something -- where the former "elaborates" so that `T: Clone` is "short" for `WF(T: Clone), T: Clone`, just as it would be in an impl. It seems useful to be able to suppress that elaboration, but I imagine that we'll almost always want it to happen, no?